src/test/java/com/google/api/codegen/DiscoveryGeneratorTestBase.java

```diff
@@ -53,6 +60,23 @@ public DiscoveryGeneratorTestBase(
     this.name = name;
     this.discoveryDocFileName = discoveryDocFileName;
     this.gapicConfigFileNames = gapicConfigFileNames;
+
+    // Look for a overrides json based on the filename of the discovery doc.
```

Comment:
This is the mechanism I've decided on to test overrides:

It looks to see if there's a file with an overrides extension matching the discovery doc and applies that to the generator.